clients/admin-ui/src/features/system/ManualSystemFlow.tsx

```diff
@@ -45,56 +47,141 @@ const ConfigureSteps = ({
 const ManualSystemFlow = () => {
   const router = useRouter();
   const dispatch = useAppDispatch();
-
   const [currentStepIndex, setCurrentStepIndex] = useState(0);
   const activeSystem = useAppSelector(selectActiveSystem);
+  const { navV2 } = useFeatures();
 
-  const goBack = () => {
+  const goBack = useCallback(() => {
     router.back();
     dispatch(setActiveSystem(undefined));
-  };
+  }, [dispatch, router]);
+
+  const handleSuccess = useCallback(
+    (system: System) => {
+      setCurrentStepIndex(currentStepIndex + 1);
+      dispatch(setActiveSystem(system));
+    },
+    [currentStepIndex, dispatch]
+  );
 
-  const handleSuccess = (system: System) => {
-    setCurrentStepIndex(currentStepIndex + 1);
-    dispatch(setActiveSystem(system));
-  };
+  const getTabs = useMemo(
+    () => () => {
+      const result: TabData[] = [];
+      STEPS.forEach((step, index) => {
+        let data: TabData | undefined;
+        switch (index) {
```

Comment:
Nitpick: I found this case statement, nested function, and array pushing much harder to follow than the pattern of jsx ternaries that was already in the page. If a JSX block like that gets too big, I'm all for pulling that into its own little component (StepTabs or something).

Reply:
Oh I see this is to slot into `DataTabs`  props. In that case maybe just extracting the function that builds this `TabData[]` into a helper function instead of the component body would do the trick.

Reply:
Issue has been resolved.